Spawn-App-iOS-SwiftUI/Views/SubComponents/TagsSubViews/TagsListView.swift

```diff
@@ -15,18 +15,85 @@ struct TagsListView: View {
     }
     
     var body: some View {
-        ScrollView {
-            VStack(alignment: .leading, spacing: 20) {
-                // Title and Time Information
-                ForEach(viewModel.friendTags) { friendTag in
-                    Text(friendTag.displayName)
+        VStack{
+            //search bar
+            searchView
+            
+            ScrollView {
+                //list of tags
+                VStack(alignment: .leading, spacing: 20) {
+                    // Title and Time Information
+                    ForEach(viewModel.friendTags) { friendTag in
+                        HStack {
+                            Text(friendTag.displayName)
+                                .foregroundColor(.white)
+                                .padding()
+                            Spacer()
+                            Image(systemName: "chevron.down")
+                                .foregroundColor(.white)
+                                .padding()
+                        }
+                        .background(friendTag.color)
+                        .cornerRadius(10)
+                        .padding(.horizontal)
+                    }
                 }
+                .padding(20)
+                .background(universalAccentColor)
+                .cornerRadius(universalRectangleCornerRadius)
             }
-            .padding(20)
-            .background(universalAccentColor)
-            .cornerRadius(universalRectangleCornerRadius)
+            
+            //add tag button
+            Button(action: {
+                // Action for adding a new tag
+            }) {
+                HStack {
+                    Image(systemName: "plus")
+                        .font(.title)
+                }
+                .frame(maxWidth: .infinity, minHeight: 50)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 10)
+                        .stroke(style: StrokeStyle(lineWidth: 2, dash: [5]))
+                        .foregroundColor(.gray)
+                )
+            }
+            //            .padding(.horizontal) // Reduces padding on the bottom
+            //            .padding(.top, 200)
+        }
+        .padding()
+        .background(
+            RoundedRectangle(cornerRadius: 30)
+                .fill(Color(hex: "#E7E7DD"))
```

Comment:
use the `universalBackgroundColor` variable, as they're the same hex code

Reply:
We want to reduce the amount of places to change this hex code in (ideally just 1--for the variable), if we ever need to change it.

Reply:
This would be especially useful if we decide to have dark/light theme, or something like that